src/core/server/http/http_server.ts

```diff
@@ -187,6 +189,33 @@ export class HttpServer {
     });
   }
 
+  private setupConditionalCompression(config: HttpConfig) {
+    if (this.server === undefined) {
+      throw new Error('Server is not created yet');
+    }
+
+    const { enabled, referrerWhitelist: list } = config.compression;
+    if (!enabled) {
+      this.log.debug('HTTP compression is disabled');
+      this.server.ext('onRequest', (request, h) => {
+        request.info.acceptEncoding = '';
+        return h.continue;
```

Comment:
Isn't this quite brutal? I think this header can be used for other things than compression?

Reply:
Not according to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding) and [RFC 7231](https://tools.ietf.org/html/rfc7231#section-3.1.2.1) -- the only valid non-compression-related directive is `identity`, which is a synonym for "no encoding".

Reply:
Oups, this and accept are not the same header.